libs/application/template-api-modules/src/lib/modules/templates/financial-statement-cemetery/financial-statement-cemetery.service.ts

```diff
@@ -0,0 +1,160 @@
+import { S3 } from 'aws-sdk'
+import { BaseTemplateApiService } from '../../base-template-api.service'
+import { Inject } from '@nestjs/common'
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import {
+  CemeteryFinancialStatementValues,
+  FinancialStatementsInaoClientService,
+  Contact,
+  ContactType,
+  DigitalSignee,
+} from '@island.is/clients/financial-statements-inao'
+import {
+  ApplicationTypes,
+  ApplicationWithAttachments as Application,
+  PerformActionResult,
+} from '@island.is/application/types'
+import { getValueViaPath } from '@island.is/application/core'
+import AmazonS3URI from 'amazon-s3-uri'
+import { TemplateApiModuleActionProps } from '../../../types'
+import * as kennitala from 'kennitala'
+import {
+  DataResponse,
+  getCurrentUserType,
+} from '../financial-statements-inao/financial-statements-inao.service'
+import {
+  BoardMember,
+  FSIUSERTYPE,
+} from '@island.is/application/templates/financial-statements-inao/types'
+import {
+  mapValuesToCemeterytype,
+  getNeededCemeteryValues,
+  mapContactsAnswersToContacts,
+  mapDigitalSignee,
+} from '../financial-statement-cemetery/mappers/mapValuesToUserType'
+import { TemplateApiError } from '@island.is/nest/problem'
+import { ApplicationApiAction } from '../../template-api.service'
+
+export type AttachmentData = {
+  key: string
+  name: string
+}
+
+export class FinancialStatementCemeteryTemplateService extends BaseTemplateApiService {
+  s3: S3
+  constructor(
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+    private financialStatementClientService: FinancialStatementsInaoClientService,
+  ) {
+    super(ApplicationTypes.FINANCIAL_STATEMENT_CEMETERY)
+    this.s3 = new S3()
+  }
+
+  private async getAttachments(application: Application): Promise<string> {
+    const attachments: Array<AttachmentData> | undefined = getValueViaPath(
+      application.answers,
+      'attachments.files',
+    ) as Array<{ key: string; name: string }>
+
+    const attachmentKey = attachments[0].key
+
+    const fileName = (
+      application.attachments as {
+        [key: string]: string
+      }
+    )[attachmentKey]
+
+    if (!fileName) {
+      return Promise.reject({})
+    }
+
+    const { bucket, key } = AmazonS3URI(fileName)
+
+    const uploadBucket = bucket
+    try {
+      const file = await this.s3
+        .getObject({
+          Bucket: uploadBucket,
+          Key: key,
+        })
+        .promise()
+      const fileContent = file.Body as Buffer
+      return fileContent.toString('base64') || ''
+    } catch (error) {
+      throw new Error('Villa kom upp við að senda umsókn')
```

Comment:
Might be an unhelpful error text, maybe, 'Villa kom upp vid að sækja viðhengi'? or something similar in English.